modules/imgproc/include/opencv2/imgproc.hpp

```diff
@@ -4757,6 +4757,73 @@ Point LineIterator::pos() const
 
 //! @endcond
 
+class CV_EXPORTS LineVirtualIterator
+{
+public:
+    /** @brief initializes the iterator
+
+    creates iterators for the line connecting pt1 and pt2
+    the line will be clipped on the image boundaries
+    the line is 8-connected or 4-connected
+    If leftToRight=true, then the iteration is always done
+    from the left-most point to the right most,
+    not to depend on the ordering of pt1 and pt2 parameters
+    */
+    LineVirtualIterator(const Size& size, Point pt1, Point pt2,
+                        int connectivity = 8, bool leftToRight = false );
```

Comment:
> leftToRight

I believe direction should be specified by `pt1`/`pt2` instead of additional parameter.
Especially, in case `leftToRight = false` doesn't mean "right-to-left".

---

Upd: `forceLeftToRight` naming makes more sense.